python/lsst/dax/obscore/obscore_exporter.py

```diff
@@ -0,0 +1,403 @@
+                # _LOG.debug("New record, records=%s", dataId.records)
+
+                record: Dict[str, Any] = {}
+
+                record["dataproduct_type"] = dataset_config.dataproduct_type
+                record["dataproduct_subtype"] = dataset_config.dataproduct_subtype
+                record["o_ucd"] = dataset_config.o_ucd
+                record["facility_name"] = self.config.facility_name
+                record["calib_level"] = dataset_config.calib_level
+                record["obs_collection"] = self.config.obs_collection
+                record["access_format"] = dataset_config.access_format
+
+                record["instrument_name"] = dataId.get("instrument")
+
+                timespan = dataId.timespan
+                if timespan is not None:
+                    t_min = cast(astropy.time.Time, timespan.begin)
+                    t_max = cast(astropy.time.Time, timespan.end)
+                    record["t_min"] = t_min.mjd
+                    record["t_max"] = t_max.mjd
+
+                region = dataId.region
+                if exposure in dataId:
+                    if (dimension_record := dataId.records[exposure]) is not None:
+                        self._exposure_records(dimension_record, record)
+                        region = self._exposure_region(dataId)
+                elif visit in dataId:
+                    if (dimension_record := dataId.records[visit]) is not None:
+                        self._visit_records(dimension_record, record)
+
+                self._region_to_columns(region, record)
+
+                if band in dataId:
+                    em_range = None
+                    if (label := dataId.get(physical_filter)) is not None:
+                        em_range = self.config.spectral_ranges.get(label)
+                    if not em_range:
+                        band_name = dataId[band]
+                        assert isinstance(band_name, str), "Band name must be string"
+                        em_range = self.config.spectral_ranges.get(band_name)
+                    if em_range:
+                        record["em_min"], record["em_max"] = em_range
+                    else:
+                        _LOG.warning("could not find spectral range for dataId=%s", dataId.full)
+                    record["em_filter_name"] = dataId["band"]
+
+                # Dictionary to use for substitutions when when formatting
+                # various strins.
+                fmt_kws: Dict[str, Any] = dict(records=dataId.records)
+                fmt_kws.update(dataId.byName())
+                fmt_kws.update(record)
+                if dataset_config.obs_id_fmt:
+                    record["obs_id"] = dataset_config.obs_id_fmt.format(**fmt_kws)
+                    fmt_kws["obs_id"] = record["obs_id"]
+
+                if self.config.use_butler_uri:
+                    try:
+                        url = self.butler.datastore.getURI(ref)
```

Comment:
Should be using `getURIs` in case of disassembly. I'm not entirely sure what should happen if something is disassembled but at least the code should understand that it might happen (for example if you ran this exporter on the pipelines_check repo).

Reply:
I think our general conclusion is not to use Butler URI, this branch (use_butler_uri==True) is probably not going to be used, maybe only for some sort of testing.

Reply:
Agreed. At the meeting today we agreed that the URI we do use would be templated and likely include the UUID.